hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/NamenodeHeartbeatService.java

```diff
@@ -482,17 +502,35 @@ private void updateHAStatusParameters(NamenodeStatusReport report) {
     }
   }
 
+  /**
+   * Evaluates whether the JMX report should be refreshed by
+   * calling the Namenode, based on the following conditions:
+   * 1. JMX Updates must be enabled.
+   * 2. The last attempt to update JMX occurred before the
+   *    configured interval (if any).
+   */
+  private boolean shouldUpdateJmx() {
+    return this.updateJmxIntervalMs >= 0
```

Comment:
Might be cleaner to read as:
```
if (this.updateJmxIntervalMs == 0)
  return false;
}
return Time.monotonicNow() - this.lastJmxUpdateAttempt > this.updateJmxIntervalMs
```

Reply:
Or < 0.

Reply:
Right, we'd have to return false if < 0, but I agree it's more readable. I made the change.